### PR TITLE
AML-1714-Consolidate_HashingService_into_single_package 

### PR DESCRIPTION
**Pull request by samuel-dev-dba**
Extended the Hashing service to handle Guid Ids

### CODE REVIEW

File: SFA.DAS.HashingService/SFA.DAS.HashingService/HashingService.cs [hunk 1]
```diff
@@ -31,5 +37,16 @@ public long DecodeValue(string id)
 
             return _hashIds.DecodeLong(id)[0];
         }
+
+        public Guid DecodeValueToGuid(string id)
+        {
+
+            if (string.IsNullOrEmpty(id?.Trim()))
```
Comment (MichaelYoung1981) [inline SFA.DAS.HashingService/SFA.DAS.HashingService/HashingService.cs]: This condition isn't tested...


Personally, I'd remove the check but if you want to keep it there should be a test.

Comment (samuel-dev-dba) [reply SFA.DAS.HashingService/SFA.DAS.HashingService/HashingService.cs]: Good spot. I kind of like it as is more descriptive, compare to any other thing that could make the code fail.  Will add the test.

Comment (samuel-dev-dba) [reply SFA.DAS.HashingService/SFA.DAS.HashingService/HashingService.cs]: Good spot.

File: SFA.DAS.HashingService/SFA.DAS.HashingService.UnitTest/HashingServiceTest/WhenHashingIds.cs [hunk 2]
```diff
@@ -52,7 +52,7 @@ public void When_Decode_HashId_IsNullOrEmpty_Should_ThrowException(string valueT
         [TestCase(333333)]
         [TestCase(444444)]
         [TestCase(0)]
-        public void Then_HashValue_Should_Equal_DecodeValue(long expectTedValue)
+        public void Then_Numric_HashValue_Should_Equal_DecodeValue(long expectTedValue)
```
Comment (MichaelYoung1981) [inline SFA.DAS.HashingService/SFA.DAS.HashingService.UnitTest/HashingServiceTest/WhenHashingIds.cs]: Just noticed typo in Numeric.

**Pull request merged**